### PR TITLE
Make IValueProvider sync, IValueProviderFactory async

### PR DESCRIPTION
**Pull request by rynowak**
/cc @dougbu - this follows the other VP change

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/IValueProviderFactory.cs [hunk 1]
```diff
@@ -12,6 +13,6 @@ public interface IValueProviderFactory
         /// </summary>
         /// <param name="context">ValueProviderFactoryContext that value provider will populate from</param>
         /// <returns>a value provider instance or null</returns>
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/IValueProviderFactory.cs]: > A `Task` that on completion returns the `<see cref="IValueProvider" />`  instance

Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/IValueProviderFactory.cs]: I don't want to live on this planet anymore

File: src/Microsoft.AspNet.Mvc.Core/ModelBinding/IEnumerableValueProvider.cs [hunk 2]
```diff
@@ -2,12 +2,11 @@
...
 
 using System.Collections.Generic;
-using System.Threading.Tasks;
 
 namespace Microsoft.AspNet.Mvc.ModelBinding
 {
     public interface IEnumerableValueProvider : IValueProvider
     {
-        Task<IDictionary<string, string>> GetKeysFromPrefixAsync(string prefix);
+        IDictionary<string, string> GetKeysFromPrefix(string prefix);
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Core/ModelBinding/IEnumerableValueProvider.cs]: Follow up PR with docs where missing?

Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/ModelBinding/IEnumerableValueProvider.cs]: I'll just do it as part of this PR

**Comment by pranavkm** [comment]
:shipit:

File: src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/IValueProviderFactory.cs (hunk 1 above)
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/IValueProviderFactory.cs]: Sounds like unconscious bias.

**Comment by rynowak** [comment]
This has gone wonky. See c2a3ac012973c5b6722398d2f1b70f403b580e1e

**Pull request closed without merging**